source/slang/slang-parameter-binding.cpp

```diff
@@ -1065,81 +1075,91 @@ static void addExplicitParameterBindings_GLSL(
-                foundSubpass = true;
-                if (foundBinding) 
-                    break;
-            }
+            info[kResInfo].resInfo = foundResInfo;
+            info[kResInfo].semanticInfo.index = glslBindingAttr->binding;
+            info[kResInfo].semanticInfo.space = glslBindingAttr->set;
         }
     }
-    else if( (foundResInfo = typeLayout->FindResourceInfo(LayoutResourceKind::SubElementRegisterSpace)) != nullptr )
+    else if(auto foundSubElementRegisterSpace = typeLayout->FindResourceInfo(LayoutResourceKind::SubElementRegisterSpace))
     {
+        foundResInfo = foundSubElementRegisterSpace;
         // Try to find `set`
         if (auto attr = varDecl.getDecl()->findModifier<GLSLBindingAttribute>())
         {
-            resInfo = foundResInfo;
+            info[kResInfo].resInfo = foundResInfo;
             if (attr->binding != 0)
             {
                 getSink(context)->diagnose(attr, Diagnostics::wholeSpaceParameterRequiresZeroBinding, varDecl.getName(), attr->binding);
             }
-            semanticInfo.index = attr->set;
-            semanticInfo.space = 0;
+            info[kResInfo].semanticInfo.index = attr->set;
+            info[kResInfo].semanticInfo.space = 0;
         }
     }
-    else if( (resInfo = typeLayout->FindResourceInfo(LayoutResourceKind::SpecializationConstant)) != nullptr )
+    else if(auto foundSpecializationConstant = typeLayout->FindResourceInfo(LayoutResourceKind::SpecializationConstant))
     {
+        info[kResInfo].resInfo = foundSpecializationConstant;
         DeclRef<Decl> varDecl2(varDecl);
 
         // Try to find `constant_id` binding
-        if(!findLayoutArg<GLSLConstantIDLayoutModifier>(varDecl2, &semanticInfo.index))
+        if(!findLayoutArg<GLSLConstantIDLayoutModifier>(varDecl2, &info[kResInfo].semanticInfo.index))
             return;
     }
 
-    // if we found resInfo, we add the explicit binding
-    if (resInfo)
+    if(auto varDeclBase = as<VarDeclBase>(varDecl))
     {
-        auto kind = resInfo->kind;
-        auto count = resInfo->count;
-        semanticInfo.kind = kind;
-
-        if (auto varDeclBase = varDecl.as<VarDeclBase>())
+        bool setABinding = false;
+        for (int i = 0; i < kMaxResCount; i++)
         {
+            auto* resInfoItem = info[i].resInfo;
+            auto& semanticInfo = info[i].semanticInfo;
+            if (!resInfoItem)
+                continue;
+        
+            auto kind = resInfoItem->kind;
+            auto count = resInfoItem->count;
+            semanticInfo.kind = kind;
+            
             addExplicitParameterBinding(context, parameterInfo, varDeclBase.getDecl(), semanticInfo, count);
-            if (foundSubpass)
-                addExplicitParameterBinding(context, parameterInfo, varDeclBase.getDecl(), subpassSemanticInfo, count);
+            setABinding = true;
         }
-        return;
+        if(setABinding)
+            return;
```

Comment:
I am not sure if it matters but the behavior here is little different from the previous code.
When `resInfo` was found, it was always returning regardless of `varDecl` can be converted to `VarDeclBase`.
But your change will return only when both conditions are true.

Reply:
I think it may matter for the language server-- I will change the logic to how it was before